File: app/src/main/java/io/tanjundang/study/knowledge/customview/CustomViewActivity.java
package io.tanjundang.study.knowledge.customview;

import android.os.Bundle;
import android.view.View;

import io.tanjundang.study.R;
import io.tanjundang.study.base.BaseActivity;

public class CustomViewActivity extends BaseActivity {
    BatteryView batteryView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
    }

    @Override
    protected void initView() {
        setContentView(R.layout.activity_custom_view);
        batteryView= (BatteryView) findViewById(R.id.batteryView);
        batteryView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                batteryView.updateColor();
            }
        });
    }

    @Override
    protected void initData() {

    }
}

File: app/src/main/java/io/tanjundang/study/knowledge/customview/CustomViewActivity.kt
package io.tanjundang.study.knowledge.customview

import android.os.Bundle
import android.view.View

import io.tanjundang.study.R
import io.tanjundang.study.base.BaseActivity

class CustomViewActivity : BaseActivity() {
    internal var batteryView: BatteryView? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
    }

    override fun initView() {
        setContentView(R.layout.activity_custom_view)
        batteryView = findViewById<View>(R.id.batteryView) as BatteryView
        batteryView!!.setOnClickListener { batteryView!!.updateColor() }
    }

    override fun initData() {

    }
}
